### PR TITLE
Add Initial Darkness Ablaze Groovy Template

### PR DESCRIPTION
**Pull request by luongthomas**
This PR introduces the first version of the Darkness Ablaze, the latest set to be released in August.

Please review to see if there is any mistakes.

### CODE REVIEW

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 1]
```diff
@@ -0,0 +1,3389 @@
...
+          text "110 damage. This Pokémon can’t attack during your next turn."
+          energyCost C, C, C
+          attackRequirement {}
+          onAttack {
+            damage 110
+          }
+        }
+      };
+      case KARRABLAST_8:
+      return basic (this, hp:HP060, type:G, retreatCost:20) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:412]: > retreatCost:20

Should be 1

Comment (luongthomas) [reply src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:412]: 😱

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 2]
```diff
@@ -0,0 +1,3389 @@
...
+          text "40 damage. "
+          energyCost G, C
+          attackRequirement {}
+          onAttack {
+            damage 40
+          }
+        }
+      };
+      case TSAREENA_16:
+      return evolution (this, from:"Steenee", hp:HP150, type:null, retreatCost:2) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:534]: > type:null

That's the wrong Pokémon. :laughing: Should be G

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 3]
```diff
@@ -0,0 +1,3389 @@
...
+          text "100+ damage. If any of your [R] Pokémon have damage counters on them, this attack does 100 more damage."
+          energyCost R, R, C
+          attackRequirement {}
+          onAttack {
+            damage 100
+          }
+        }
+      };
+      case TORCHIC_23:
+      return basic (this, hp:HP060, type:R, retreatCost:null) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:666]: > retreatCost:null

Should be 1

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 4]
```diff
@@ -0,0 +1,3389 @@
...
+          text "10 damage. "
+          energyCost R
+          attackRequirement {}
+          onAttack {
+            damage 10
+          }
+        }
+      };
+      case COMBUSKEN_24:
+      return evolution (this, from:"Torchic", hp:HP090, type:R, retreatCost:null) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:686]: > retreatCost:null

Should be 1

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 5]
```diff
@@ -0,0 +1,3389 @@
...
+        weakness W
+        move "Fire Fang", {
+          text "30 damage. Your opponent’s Active Pokémon is now Burned."
+          energyCost R, C
+          attackRequirement {}
+          onAttack {
+            damage 30
+          }
+        }
+      };
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:733]: If the following card is the correct one, missing the second attack.

[Heatran](https://www.pokebeach.com/news/0420/038073_P_HIDORAN-2.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 6]
```diff
@@ -0,0 +1,3389 @@
...
+          text "30 damage. "
+          energyCost R, C
+          attackRequirement {}
+          onAttack {
+            damage 30
+          }
+        }
+      };
+      case PANSEAR_28:
+      return basic (this, hp:HP060, type:R, retreatCost:null) {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy]: > retreatCost:null

Should be 1

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 7]
```diff
@@ -0,0 +1,3389 @@
...
+          text "120+ damage. If your opponent’s Active Pokémon is a Pokémon VMAX, this attack does 120 more damage."
+          energyCost D, C, C, C
+          attackRequirement {}
+          onAttack {
+            damage 120
+          }
+        }
+      };
+      case ETERNATUS_VMAX_128:
+      return evolution (this, from:"Eternatus V", hp:HP340, type:D, retreatCost:2) {
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2516]: Should be retreat cost 3, not 2.

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy (hunk 5 above)
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:741]: Good catch!

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy (hunk 7 above)
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2516]: Nice catch!

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 8]
```diff
@@ -0,0 +1,3394 @@
...
+          onAttack {
+            damage 80
+          }
+        }
+      };
+      case HOOPA_121:
+      return basic (this, hp:HP120, type:D, retreatCost:2) {
+        weakness G
+        move "Assault Gate", {
+          text "90 damage. If this Pokémon wasn''t moved from your Bench to the Active Spot during your turn, this attack does nothing. This attack isn’t affected by Weakness."
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2403]: (minor)
~~wasn''t~~ -> wasn't

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 9]
```diff
@@ -0,0 +1,3394 @@
...
+          onAttack {
+            damage 130
+          }
+        }
+      };
+      case ELECTRIKE_57:
+      return basic (this, hp:HP060, type:L, retreatCost:1) {
+        weakness F
+        move "Thunderbolt", {
+          text "60 damage. Discard all Energy from this Pokémon."
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:1267]: I believe damage for this attack is 50

[Electrike](https://www.pokebeach.com/news/0520/038168_P_RAKURAI.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 10]
```diff
@@ -0,0 +1,3394 @@
...
+        move "Bite", {
+          text "30 damage. "
+          energyCost C
+          attackRequirement {}
+          onAttack {
+            damage 30
+          }
+        }
+        move "Power Blast", {
+          text "130 damage. Discard an Energy attached to this Pokémon."
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2057]: Should say "Discard a [F] Energy attached to this Pokémon."

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 11]
```diff
@@ -0,0 +1,3394 @@
...
+      case TRAPINCH_92:
+      return basic (this, hp:HP060, type:F, retreatCost:1) {
+        weakness G
+        move "Sand Impulse", {
+          text "20 damage. This attack does 10 damage to each of your opponent’s Benched Pokémon. (Don’t apply Weakness and Resistance.)"
+          energyCost F
+          attackRequirement {}
+          onAttack {
+            damage 20
+          }
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:1916]: This is the attack from Vibrava. Correct one is:

> [F] Land’s Pulse: 10+ damage. If there is any Stadium card in play, this attack does 10 more damage.

```suggestion
        move "Land's Pulse", {
          text "10+ damage. If there is any Stadium card in play, this attack does 10 more damage."
          energyCost F
          attackRequirement {}
          onAttack {
            damage 10
          }
```

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 12]
```diff
@@ -0,0 +1,3394 @@
...
+        }
+        move "TBA", {
+          text "100 damage. TBA"
+          energyCost C, C, C
+          attackRequirement {}
+          onAttack {
+            damage 100
+          }
+        }
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:1827]: Not reaaaally relevant but maybe add a comment about rewriting this whole card once revealed?

```suggestion
      case GALARIAN_FARFETCH_D_87: //TODO: Get data on this card.
      return basic (this, hp:HP010, type:C, retreatCost:2) {
        weakness Y
        resistance F, MINUS30
        bwAbility "TBA", {
          text "TBA"
          actionA {
          }
        }
        move "TBA", {
          text "100 damage. TBA"
          energyCost C, C, C
          attackRequirement {}
          onAttack {
            damage 100
          }
        }
```

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 13]
```diff
@@ -0,0 +1,3394 @@
...
+          onAttack {
+            damage 250
+          }
+        }
+      };
+      case TRAPINCH_92:
+      return basic (this, hp:HP060, type:F, retreatCost:1) {
+        weakness G
+        move "Sand Impulse", {
+          text "20 damage. This attack does 10 damage to each of your opponent’s Benched Pokémon. (Don’t apply Weakness and Resistance.)"
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:1911]: Looks like this is 10+ damage

[Trapinch](https://www.pokebeach.com/news/0520/038182_P_NAKKURA.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 14]
```diff
@@ -0,0 +1,3394 @@
...
+      return evolution (this, from:"Grimer", hp:HP140, type:D, retreatCost:4) {
+        weakness F
+        move "Triple Poison", {
+          text "Your opponent’s Active Pokémon is now Poisoned. Put 3 damage counters for this Poison between turns."
+          energyCost D
+          attackRequirement {}
+          onAttack {
+
+          }
+        }
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2115]: Look to be missing the second move for Muk too

[Muk](https://www.pokebeach.com/news/0420/038109_P_BETOBETON-1.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 15]
```diff
@@ -0,0 +1,3394 @@
...
+      return evolution (this, from:"Scraggy", hp:HP120, type:D, retreatCost:2) {
+        weakness G
+        move "Corner", {
+          text "30 damage. The Defending Pokémon can’t retreat during your opponent’s next turn."
+          energyCost D
+          attackRequirement {}
+          onAttack {
+            damage 30
+          }
+        }
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2357]: Missing second attack

[Scrafty](https://www.pokebeach.com/news/0520/038196_P_ZURUZUKIN.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 16]
```diff
@@ -0,0 +1,3394 @@
...
+          onAttack {
+            damage 20
+          }
+        }
+      };
+      case MALAMAR_120:
+      return evolution (this, from:"Inkay", hp:HP120, type:D, retreatCost:3) {
+        weakness G
+        move "Eerie Wave", {
+          text "Your opponent’s Active Pokémon is now Confused."
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2383]: Does 50 damage

[Malamar](https://www.pokebeach.com/news/0520/038198_P_KARAMANERO.jpg)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 17]
```diff
@@ -0,0 +1,3394 @@
...
+      return pokemonTool (this) {
+        text "If your opponent’s Active Pokémon is the same type as the Weakness of the Pokémon this card is attached to, the attacks of this Pokémon do 30 more damage to your opponent’s Active Pokémon."
+        onPlay {reason->
+        }
+        onRemoveFromPlay {
+        }
+        allowAttach {to->
+        }
+      };
+      case ADVERSITY_GLOVES_165:
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:3208]: Duplicate

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 18]
```diff
@@ -0,0 +1,3410 @@
...
+  SIMISAGE_7 ("Simisage", 7, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _GRASS_]),
+  KARRABLAST_8 ("Karrablast", 8, Rarity.HOLORARE, [POKEMON, BASIC, _GRASS_]),
+  SHELMET_9 ("Shelmet", 9, Rarity.HOLORARE, [POKEMON, BASIC, _GRASS_]),
+  ACCELGOR_10 ("Accelgor", 10, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _GRASS_]),
+  ROWLET_11 ("Rowlet", 11, Rarity.HOLORARE, [POKEMON, BASIC, _GRASS_]),
+  DARTRIX_12 ("Dartrix", 12, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _GRASS_]),
+  DECIDUEYE_13 ("Decidueye", 13, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE2, _GRASS_]),
+  BOUNSWEET_14 ("Bounsweet", 14, Rarity.HOLORARE, [POKEMON, BASIC, _GRASS_]),
+  STEENEE_15 ("Steenee", 15, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _GRASS_]),
+  TSAREENA_16 ("Tsareena", 16, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE2]),
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:57]: I assume this is important
```suggestion
  TSAREENA_16 ("Tsareena", 16, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE2, _GRASS_]),
```

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 19]
```diff
@@ -0,0 +1,3410 @@
...
+  SCRAGGY_117 ("Scraggy", 117, Rarity.HOLORARE, [POKEMON, BASIC, _DARKNESS_]),
+  SCRAFTY_118 ("Scrafty", 118, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _DARKNESS_]),
+  INKAY_119 ("Inkay", 119, Rarity.HOLORARE, [POKEMON, BASIC, _DARKNESS_]),
+  MALAMAR_120 ("Malamar", 120, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _DARKNESS_]),
+  HOOPA_121 ("Hoopa", 121, Rarity.HOLORARE, [POKEMON, BASIC, _DARKNESS_]),
+  NICKIT_122 ("Nickit", 122, Rarity.HOLORARE, [POKEMON, BASIC, _DARKNESS_]),
+  NICKIT_123 ("Nickit", 123, Rarity.HOLORARE, [POKEMON, BASIC, _DARKNESS_]),
+  THIEVUL_124 ("Thievul", 124, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, _DARKNESS_]),
+  GRIMMSNARL_V_125 ("Grimmsnarl V", 125, Rarity.HOLORARE, [POKEMON, BASIC, POKEMON_V, _DARKNESS_]),
+  GRIMMSNARL_VMAX_126 ("Grimmsnarl VMAX", 126, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, VMAX]),
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:167]: Here too
```suggestion
  GRIMMSNARL_VMAX_126 ("Grimmsnarl VMAX", 126, Rarity.HOLORARE, [POKEMON, EVOLUTION, STAGE1, VMAX, _DARKNESS_]),
```

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 20]
```diff
@@ -0,0 +1,3410 @@
...
+      case BIG_PARASOL_166:
+      return pokemonTool (this) {
+				text "As long as the Pokemon this card is attached to is your Active Pokemon, prevent all effects of your opponent’s attacks, except damage, done to all of your Pokemon (existing effects are not removed)."
+				onPlay {reason->
+				}
+				onRemoveFromPlay {
+				}
+				allowAttach {to->
+				}
+			};
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:3217]:
```suggestion
      case BIG_PARASOL_166:
      return pokemonTool (this) {
        text "As long as the Pokémon this card is attached to is your Active Pokémon, prevent all effects of your opponent’s attacks, except damage, done to all of your Pokémon (existing effects are not removed)."
        onPlay {reason->
        }
        onRemoveFromPlay {
        }
        allowAttach {to->
        }
      };
```

**Pull request merged**